src/Platforms/CUDA/CUDAruntime.hpp

```diff
@@ -25,6 +25,17 @@
 #include "Platforms/ROCm/cuda2hip.h"
 #endif
 
+// Work around for clang using gcc/glibc 12 include files in CUDA mode.
+// Clang was fixed to recognize __noinline__ as an attribute,
+// but the CUDA include files still have a define for __noinline__.
+// The attribute started to be used in gcc 12 include files, which expands
+// to __attribute__((__attribute__((noinline)))), which the compiler rejects.
+// See https://github.com/NVIDIA/thrust/issues/1703 and
+//     https://github.com/llvm/llvm-project/issues/57544
+#if defined(__clang__) && (_GLIBCXX_RELEASE >= 12)
+#undef __noinline__
+#endif
```

Comment:
Is this workaround recommended by someone? Do you have a link?

Reply:
It's mentioned in the LLVM issue link in the text.  Now that I read the LLVM issue link further, it looks like there's a workaround in clang itself.  I don't know why that workaround isn't working with my clang install.

Reply:
I think the reason the llvm workaround isn't working is because CUDADeviceManager.cpp is C++ file, not a CUDA file.